### PR TITLE
feat: adding traces support (MAPCO-3893)

### PR DESCRIPTION
**Pull request by almog8k**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✔                                                                       |

Further  information:
Adding traces support.

### CODE REVIEW

**Review by alebinson — APPROVED** [review]
Looks good (decoration principles are nice and clear)

File: config/custom-environment-variables.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
     },
     "tracing": {
       "enabled": "TELEMETRY_TRACING_ENABLED",
-      "url": "TELEMETRY_TRACING_URL"
+      "url": "TELEMETRY_TRACING_URL",
```
Comment (asafMasa) [inline config/custom-environment-variables.json:18]: add to default config

File: src/jobs/models/jobManager.ts [hunk 2]
```diff
@@ -76,26 +82,30 @@ export class JobManager {
...
 
+  @withSpanAsyncV4
   public async isResettable(req: IJobsParams): Promise<IIsResettableResponse> {
     const jobId = req.jobId;
     const repo = await this.getRepository();
     const isResettable = await repo.isJobResettable(jobId);
     return { jobId, isResettable };
   }
 
+  @withSpanAsyncV4
```
Comment (asafMasa) [inline src/jobs/models/jobManager.ts:108]: add for getAvailableActions, isAbortable

File: src/taskManagement/models/taskManagementManger.ts [hunk 3]
```diff
@@ -14,8 +16,13 @@ export class TaskManagementManager {
...
   private taskRepository?: TaskRepository;
 
-  public constructor(@inject(SERVICES.LOGGER) private readonly logger: Logger, private readonly connectionManager: ConnectionManager) {}
+  public constructor(
+    @inject(SERVICES.LOGGER) private readonly logger: Logger,
+    @inject(SERVICES.TRACER) public readonly tracer: Tracer,
+    private readonly connectionManager: ConnectionManager
+  ) {}
 
+  @withSpanAsyncV4
```
Comment (asafMasa) [inline src/taskManagement/models/taskManagementManger.ts:25]: remove

**Review by asafMasa — CHANGES_REQUESTED** [review]

**Review by asafMasa — APPROVED** [review]

**Review by alebinson — APPROVED** [review]
looks good

**Review by netanelC — APPROVED** [review]

**Pull request merged**